src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferIsEmptyOverAnyTests.cs

```diff
@@ -0,0 +1,260 @@
+    Return Not {|#0:array.Any()|}
+End Function";
+            const string fixedCode = @"
+Public Function IsEmpty(array As ImmutableArray(Of Integer)) As Boolean
+    Return array.IsEmpty
+End Function";
+
+            return VerifyVB.VerifyCodeFixAsync(string.Format(VbTemplate, code), ExpectedDiagnostic, string.Format(VbTemplate, fixedCode));
+        }
+
+        [Fact]
+        public Task DontWarnOnChainedLinqWithAnyAsync()
+        {
+            const string code = @"
+public bool HasContents(ImmutableArray<int> array) {
+    return array.Select(x => x).Any();
+}";
+
+            return VerifyCS.VerifyAnalyzerAsync(string.Format(CSharpTemplate, code));
+        }
+
+        [Fact]
+        public Task VbDontWarnOnChainedLinqWithAnyAsync()
+        {
+            const string code = @"
+Public Function HasContents(array As ImmutableArray(Of Integer)) As Boolean
+    Return array.Select(Function(x) x).Any()
+End Function";
+
+            return VerifyVB.VerifyAnalyzerAsync(string.Format(VbTemplate, code));
+        }
+
+        [Fact]
+        public Task DontWarnOnAnyWithPredicateAsync()
+        {
+            const string code = @"
+public bool HasContents(ImmutableArray<int> array) {
+    return array.Any(x => x > 5);
+}";
+
+            return VerifyCS.VerifyAnalyzerAsync(string.Format(CSharpTemplate, code));
+        }
+
+        [Fact]
+        public Task VbDontWarnOnAnyWithPredicateAsync()
+        {
+            const string code = @"
+Public Function HasContents(array As ImmutableArray(Of Integer)) As Boolean
+    Return array.Any(Function(x) x > 5)
+End Function";
+
+            return VerifyVB.VerifyAnalyzerAsync(string.Format(VbTemplate, code));
+        }
+
+        [Fact]
+        public Task TestQualifiedCallAsync()
+        {
+            const string code = @"
+public bool HasContents(ImmutableArray<int> array) {
+    return {|#0:Enumerable.Any(array)|};
+}";
+            const string fixedCode = @"
+public bool HasContents(ImmutableArray<int> array) {
+    return !array.IsEmpty;
+}";
+
+            return VerifyCS.VerifyCodeFixAsync(string.Format(CSharpTemplate, code), ExpectedDiagnostic, string.Format(CSharpTemplate, fixedCode));
+        }
+
+        [Fact]
+        public Task VbTestQualifiedCallAsync()
+        {
+            const string code = @"
+Public Function HasContents(array As ImmutableArray(Of Integer)) As Boolean
+    Return {|#0:Enumerable.Any(array)|}
+End Function";
+            const string fixedCode = @"
+Public Function HasContents(array As ImmutableArray(Of Integer)) As Boolean
+    Return Not array.IsEmpty
+End Function";
+
+            return VerifyVB.VerifyCodeFixAsync(string.Format(VbTemplate, code), ExpectedDiagnostic, string.Format(VbTemplate, fixedCode));
+        }
+
+        [Fact]
+        public Task TestFullyQualifiedCallAsync()
+        {
+            const string code = @"
+public bool HasContents(ImmutableArray<int> array) {
+    return {|#0:System.Linq.Enumerable.Any(array)|};
```

Comment:
ImmutableArray.Any() just checks [`immutableArray.Length > 0;`](https://source.dot.net/#System.Collections.Immutable/System/Linq/ImmutableArrayExtensions.cs,99), so probably we should not warn on this. I wonder if we should flag only if the IEnumerable.Any() extension method is used CC @stephentoub

Reply:
Got a response that we want to keep flagging it, if not for perf then for clarity.